File: code/cf/edu39/3.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    string str;
    cin>>str;
    int cnt=0;
    for(int i=0;i<str.size();i++)
    {
        if(cnt==25)
        {
            str[i]='z';
            break;
        }
        if(str[i]=='a'+cnt || str[i]+1=='a'+cnt)
        {
            str[i]='a'+cnt;
            cnt++;
        }
    }
    if(cnt==25) cout << str;
    else cout << "-1";
    return 0;
}

File: code/basic/nowcoder/pku_kaoyan/3.java
import java.math.*;
import java.util.*;


public class qq
{
    public static void main(String[] args)
    {
        Scanner cin = new Scanner(System.in);
        while(cin.hasNext())
        {
            Stack<BigInteger> S = new Stack<BigInteger>();
            BigInteger num = cin.nextBigInteger();
            if(num.equals(BigInteger.valueOf(0)))
            {
                System.out.println("0");
                continue;
            }
            int cnt = 0;
            while(!num.equals(BigInteger.valueOf(0)))
            {
                //System.out.print(num.remainder(BigInteger.valueOf(2)));
                S.push(num.remainder(BigInteger.valueOf(2)));
                num = num.divide(BigInteger.valueOf(2));
            }
            while(!S.empty())
                System.out.print(S.pop());
            System.out.println();
        }
    }
}
